src/libcadet/model/reaction/MichaelisMentenReaction.cpp

```diff
@@ -28,313 +28,584 @@
- * @details Implements the Michaelis-Menten kinetics: \f[ \begin{align}
- *              S \nu,
- *          \end{align} \f]
- *          where \f$ S \f$ is the stoichiometric matrix and the fluxes are given by
- *          \f[ \begin{align}
- *              \nu_i = \frac{\mu_{\mathrm{max},i} c_S}{k_{\mathrm{MM},i} + c_S}.
- *          \end{align} \f]
- *          The substrate component \f$ c_S \f$ is identified by the index of the
- *          first negative entry in the stoichiometry of this reaction.
- *			In addition, the reaction might be inhibited by other components. In this
- *          case, the flux has the form
- *          \f[ \begin{align}
- *              \nu_i = \frac{\mu_{\mathrm{max},i} c_S}{k_{\mathrm{MM},i} + c_S} \cdot \frac{1}{1 + \sum_i \frac{1+ k_{\mathrm{I},i,j}}{k_{\mathrm{I},i,j} + c_{\mathrm{I},j}}}.
- *          \end{align} \f]
- *          The value of \f$ k_{\mathrm{I},i,j} \f$ decides whether component \f$ j \f$
- *          inhibits reaction \f$ i \f$. If \f$ k_{\mathrm{I},i,j} \leq 0 \f$, the component
- *          does not inhibit the reaction.
- *          Only reactions in liquid phase are supported (no solid phase or cross-phase reactions).
- * @tparam ParamHandler_t Type that can add support for external function dependence
- */
+    * @brief Defines a Multi-Substrate Michaelis-Menten reaction with complex inhibition
+    * @details Implements the extended Michaelis-Menten kinetics with multiple types of inhibition:
+    *          \f[ \begin{align}
+    *              S \nu,
+    *          \end{align} \f]
+    *          where \f$ S \f$ is the stoichiometric matrix and the fluxes are given by
+    *          \f[ \begin{align}
+    *              \nu_i = v_{\max} \prod_{j=1}^{N} \left( \frac{c_j}{K_{m,j} (1 + \sum_{k=1}^{N} \tilde{K}_{i,j,k} c_k) + c_j \cdot (1 + \sum_{k=1}^{N} K_{i,j,k} c_k)} \right)
+    *          \end{align} \f]
+    *          Substrate components \f$ c_j \f$ are identified by negative entries in the stoichiometry matrix.
+    *          This model supports:
+    *          - Competitive inhibition via \f$ \tilde{K}_{i,j,k} \f$ (MM_KIC)
+    *          - Uncompetitive inhibition via \f$ K_{i,j,k} \f$ (MM_KIUC)
+    *          - Non-competitive inhibition through combination of both parameters
+    * @tparam ParamHandler_t Type that can add support for external function dependence
+    */
 template <class ParamHandler_t>
-class MichaelisMentenReactionBase : public DynamicReactionModelBase
+class MichaelisMentenBase : public DynamicReactionModelBase
 {
 public:
+    MichaelisMentenBase() { }
+    virtual ~MichaelisMentenBase() CADET_NOEXCEPT { }
 
-	MichaelisMentenReactionBase() : _idxSubstrate(0) { }
-	virtual ~MichaelisMentenReactionBase() CADET_NOEXCEPT { }
+    static const char* identifier() { return ParamHandler_t::identifier(); }
+    virtual const char* name() const CADET_NOEXCEPT { return ParamHandler_t::identifier(); }
```

Comment:
a general remark, we seem to be mixing tabs, as well as spaces for indentation. This should be more consistent. See also #99